main.js

```diff
@@ -133,6 +153,35 @@ define(function (require, exports, module) {
         };
     }
     
+    /**
+     * Apply the whitespace colors.
+     * This does not overwrite styles already defined by a theme.
+     */
+    function _applyColors() {
+        // Compile the CSS from the template
+        var userColors = _preferences.get("colors"),
+            template   = _.template(stylesTemplate),
+            css = template({
+                "empty": userColors.light.empty,
+                "leading": userColors.light.leading,
+                "trailing": userColors.light.trailing,
+                "whitespace": userColors.light.whitespace,
+                
+                "darkEmpty": userColors.dark.empty,
+                "darkLeading": userColors.dark.leading,
+                "darkTrailing": userColors.dark.trailing,
+                "darkWhitespace": userColors.dark.whitespace
+            });
+
+        // If the styles do not exist, apply to the DOM
+        // Otherwise, update the existing element
+        if (!_styleInline) {
+            _styleInline = ExtensionUtils.addEmbeddedStyleSheet(css);
+        } else {
+            $(_styleInline).text(css);
```

Comment:
Sounds good :)  This is how I have done it in brackets.

Definition:
https://github.com/adobe/brackets/blob/master/src/view/ThemeManager.js#L43

Then you just use it:
https://github.com/adobe/brackets/blob/master/src/view/ThemeManager.js#L242